packages/next/server/next-server.ts

```diff
@@ -313,6 +314,18 @@ export default class Server {
     res: ServerResponse,
     parsedUrl?: UrlWithParsedQuery
   ): Promise<void> {
+    const urlParts = (req.url || '').split('?')
+    const urlNoQuery = urlParts[0]
+
+    if (urlNoQuery?.match(/(\\|\/\/)/)) {
+      const cleanUrl = normalizeRepeatedSlashes(req.url!)
+      res.setHeader('Location', cleanUrl)
+      res.setHeader('Refresh', `0;url=${cleanUrl}`)
+      res.statusCode = 308
+      res.end(cleanUrl)
+      return
+    }
```

Comment:
```suggestion
    const originalUrl = req.url || ''
    const cleanUrl = normalizeRepeatedSlashes(originalUrl)

    if (originalUrl !== cleanUrl) {
      res.setHeader('Location', cleanUrl)
      res.setHeader('Refresh', `0;url=${cleanUrl}`)
      res.statusCode = 308
      res.end(cleanUrl)
      return
    }
```

How about this?

Reply:
The `normalizeRepeatedSlashes` util expects the query to not be included as we only want to normalize the path here since normalizing query items could break some behavior.